sdcm/tester.py

```diff
@@ -2552,6 +2553,9 @@ def tearDown(self):
         self.log.info('TearDown is starting...')
         self.stop_timeout_thread()
         self.stop_event_analyzer()
+        if self.params.get('cluster_backend') == "gce":
```

Comment:
I'm talking about replace the thread we currently have, with a more accurate information (which include all system events),
and not depended on the connectivity to the machine. I don't think it add much complexity and risk, and you can use the filter API better, since you are doing for a specific instance.

Reply:
I couldn't find any other teardown - @fgelcer can you point it?

Reply:
ah, there is this decorator (https://github.com/scylladb/scylla-cluster-tests/blob/bbd367eda7f15831c80f25d628c600b60a7c63d3/sdcm/tester.py#L132), but i see that in the end, it calls the regular `tearDown`, so never mind.

Reply:
anyway, this solution had a flaw - when jenkins times out we wouldn't get `hostError` events. I've followed @fruch advice to embed it into spot_termination thread.